modules/rocksdb/src/xtdb/rocksdb.clj

```diff
@@ -24,31 +24,45 @@
 
 (defrecord RocksKvIterator [^RocksIterator i]
   kv/KvIterator
-  (seek [this k]
+  (seek [_ k]
     (.seek i (mem/direct-byte-buffer k))
     (iterator->key i))
 
-  (next [this]
-    (.next i)
-    (iterator->key i))
+  (next [_]
+    (when (.isValid i)
```

Comment:
IIUC callers shouldn't be calling `next` if the iterator isn't valid, and I'm not sure what performance impact adding this on every call will have - do you know which caller is causing the issue here?

Reply:
I think we need to _always_ be checking `.isValid` before attempting to call `.next`. Based on:

1. https://github.com/facebook/rocksdb/issues/5611#issuecomment-1097435163
2. the "REQUIRES: isValid()" docstring https://javadoc.io/doc/org.rocksdb/rocksdbjni/latest/org/rocksdb/RocksIteratorInterface.html#next--
3.  https://github.com/facebook/rocksdb/issues/9306#issuecomment-997551091
4.  https://github.com/facebook/rocksdb/wiki/Iterator#error-handling

I can reproduce a JVM crash simply by changing the definition of the Rocks module with a bunch of extra seeks (and running a basic test in kv_test.clj):
```clojure
(defrecord RocksKvIterator [^RocksIterator i]
  kv/KvIterator
  (seek [this k]
    (.seek i (mem/direct-byte-buffer k))
    (iterator->key i))

  (next [this]
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (.next i)
    (iterator->key i))

  (prev [this]
    (.prev i)
    (iterator->key i))
 ...
 ```

It's not been a problem in practice only because in normal operation `.next` will keep returning things (i.e. continue being valid) across the codec boundaries, so this could only ever show up when calling `.next` on the final entry(s) in the final codec keyspace (which is the stats index, as per https://github.com/xtdb/xtdb/blob/7ec4828f48b3fb97ffd1baf4eda22857c9fcb48d/core/src/xtdb/codec.clj#L60